doi-transformer/src/test/java/no/unit/nva/doi/transformer/CrossRefConverterTest.java

```diff
@@ -555,7 +553,58 @@ public void toPublicationSetsIsbnWhenCrossrefDocumentIsTypeBook()
                 .map(Isxn::getValue)
                 .map(ISBN_VALIDATOR::validate)
                 .collect(Collectors.toSet());
-        assertEquals(actualValues, expectedValues);
+
+        assertEquals(expectedValues, actualValues);
+    }
+
+    @Test
+    @DisplayName("toPublication filters ISBN error and continues")
+    public void toPublicationFiltersIsbnErrorAndContinues()
+            throws InvalidIssnException, InvalidIsbnException, UnsupportedDocumentTypeException {
+        CrossRefDocument sampleBook = sampleBook();
+
+        IsxnType type = IsxnType.PRINT;
+        Isxn printIsbnA = sampleIsxn(type, VALID_ISBN_A);
+        Isxn printIsbnB = sampleIsxn(type, INVALID_ISBN);
+        List<Isxn> isbns = Arrays.asList(printIsbnA, printIsbnB);
+        sampleBook.setIsbnType(isbns);
+
+        Book actualPublicationContext = (Book) toPublication(sampleBook).getEntityDescription()
+                .getReference()
+                .getPublicationContext();
+        Set<String> actualValues = new HashSet<>(actualPublicationContext.getIsbnList());
+        Set<String> expectedValues = isbns.stream()
+                .map(Isxn::getValue)
+                .map(ISBN_VALIDATOR::validate)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
+
+        assertEquals(expectedValues, actualValues);
+    }
+
+
+    @Test
+    @DisplayName("toPublication filters ISSN error and continues")
+    public void toPublicationFiltersIssnErrorAndContinues()
```

Comment:
refactor and re-use the above suggested methods if possible to make the tests readable.